src/application/common/logger.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as Logger from 'bunyan';
+
+const ILogger = Symbol("ILogger");
+
+interface ILogger {
+    addStream(stream: Logger.Stream): Promise<void>;
+    addSerializers(serializers: Logger.Serializers | Logger.StdSerializers): Promise<void>;
+    child(options: Logger.LoggerOptions, simple?: boolean): Promise<Logger>;
+    child(obj: Object, simple?: boolean): Promise<Logger>;
+    reopenFileStreams(): Promise<void>;
+
+    level(): Promise<number>;
+    level(value: number | string): Promise<void>;
+    levels(name: number | string, value: number | string): Promise<Promise<void>>;
+    fields: Promise<any>;
+    src: Promise<boolean>;
+    trace(): Promise<boolean>;
+    trace(error: Error, format?: any, ...params: any[]): Promise<Promise<void>>;
+    trace(buffer: Buffer, format?: any, ...params: any[]): Promise<Promise<void>>;
+    trace(format: string | number, ...params: any[]): Promise<Promise<void>>;
+    trace(obj: Object, format?: any, ...params: any[]): Promise<Promise<void>>;
+    debug(): Promise<boolean>;
+    debug(error: Error, format?: any, ...params: any[]): Promise<void>;
+    debug(buffer: Buffer, format?: any, ...params: any[]): Promise<void>;
+    debug(format: string | number, ...params: any[]): Promise<void>;
+    debug(obj: Object, format?: any, ...params: any[]): Promise<void>;
+    info(): Promise<boolean>;
+    info(error: Error, format?: any, ...params: any[]): Promise<void>;
+    info(buffer: Buffer, format?: any, ...params: any[]): Promise<void>;
+    info(format: string | number, ...params: any[]): Promise<void>;
+    info(obj: Object, format?: any, ...params: any[]): Promise<void>;
+    warn(): Promise<boolean>;
+    warn(error: Error, format?: any, ...params: any[]): Promise<void>;
+    warn(buffer: Buffer, format?: any, ...params: any[]): Promise<void>;
+    warn(format: string | number, ...params: any[]): Promise<void>;
+    warn(obj: Object, format?: any, ...params: any[]): Promise<void>;
+    error(): Promise<boolean>;
+    error(error: Error, format?: any, ...params: any[]): Promise<void>;
+    error(buffer: Buffer, format?: any, ...params: any[]): Promise<void>;
+    error(format: string | number, ...params: any[]): Promise<void>;
+    error(obj: Object, format?: any, ...params: any[]): Promise<void>;
+    fatal(): Promise<boolean>;
+    fatal(error: Error, format?: any, ...params: any[]): Promise<void>;
+    fatal(buffer: Buffer, format?: any, ...params: any[]): Promise<void>;
+    fatal(format: string | number, ...params: any[]): Promise<void>;
+    fatal(obj: Object, format?: any, ...params: any[]): Promise<void>;
+}
+
+export { ILogger, Logger }
```

Comment:
Well it needs to be visible to be able to create the bunyan logger in the backend, cache the proper levels in the frontend and use the bunyan types where needed...
I could have reimported bunyan in all the places that I needed it rerectly but it seemed redundant..
Another optino would be to export it as BunyanLogger maybe ?

But oviously we should not use Logger directly to log.